### PR TITLE
reduce GHA log output

### PR DESCRIPTION
**Pull request by pmeier**
Currently opening the logs of a GHA unittest job is really sluggish compared to CircleCI. The reason for this is that in the new workflows we only have a single step rather than multiple ones. Meaning, while CircleCI just needs to load a portion of the total output, GHA needs to load everything. (and that is leaving out that GHA in general seems to be a lot slower at that).

While not a fix, this PR addresses this issue:

1. There is no need for install progress to be logged in a non-interactive environment. Especially progress bars create an enormous amount of lines without any real benefit. Thus, we set the respective flags for `conda` and `pip` to install "quietly" without touching the meaningful information.
2. Our test are run with the `-v` flag. This means, every test creates a single line in the log output, whereas without we only get a single character per test. Thus depending on the console width (I think it is 80 in CI), we create roughly 60-80x more lines than necessary. In some cases GHA even truncates the displayed logs (#6587), so one has to dig into the raw ones, which are harder to parse. On CircleCI this is not as big of a problem since they have the nifty "Tests" tab that show all failed tests without going manually through the logs.

cc @seemethere

### CODE REVIEW

**Comment by pmeier** [comment]
PR: 1635 lines https://github.com/pytorch/vision/actions/runs/4194550278/jobs/7272863693#step:10:1636

main: 33352 lines https://github.com/pytorch/vision/actions/runs/4192967970/jobs/7269240560#step:10:33352

So overall a roughly 20x reduction

**Comment by NicolasHug** [comment]
I'd really prefer keeping that kind of verboe pytest output if we can

image

File: .github/workflows/test-linux-cpu.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
         fi
 
         # Create Conda Env
-        conda create -yp ci_env python="${PYTHON_VERSION}" numpy libpng jpeg scipy
+        conda create -yp ci_env --quiet python="${PYTHON_VERSION}" numpy libpng jpeg scipy
```
Comment (NicolasHug) [inline .github/workflows/test-linux-cpu.yml:42]: this is going to be very frustrating when we have env issues. So... once a week?
😔😔😔😔

Comment (pmeier) [reply .github/workflows/test-linux-cpu.yml:42]: This only affects the progress bar:

```bash
$ conda create --help | grep -e --quiet
  -q, --quiet           Do not display progress bar.
```

The actual information about the environment, i.e what is installed / upgraded / downgraded is still there: https://github.com/pytorch/vision/actions/runs/4194550278/jobs/7272863693#step:10:168

**Review by vfdev-5 — APPROVED** [review]
Thanks @pmeier !

**Review by NicolasHug — APPROVED** [review]
Even though I'd really prefer not to, I'll approve because on top of GA being frustratingly slow, it has now started to truncate the test logs as in https://github.com/pytorch/vision/actions/runs/4234487154/jobs/7356840736. This makes the tests logs impossible to see without additional clicks + we lose all formatting which makes them very hard to read.

@seemethere @osalpekar please let me know if you're not the right POC for that, but the move to GA has led to various significant decrease in workflow qualities that I hope we can address soon.

**Comment by osalpekar** [comment]
@NicolasHug We're the POC's for that. Would love to set up some time and learn about your pain points with GHA CI jobs, and how the user experience can be better.

File: .github/workflows/test-linux-cpu.yml [hunk 2]
```diff
@@ -39,9 +39,9 @@ jobs:
         fi
 
         # Create Conda Env
-        conda create -yp ci_env python="${PYTHON_VERSION}" numpy libpng jpeg scipy 'ffmpeg<4.3'
+        conda create -yp ci_env --quiet python="${PYTHON_VERSION}" numpy libpng jpeg scipy 'ffmpeg<4.3'
         conda activate /work/ci_env
-
+
```
Comment (vfdev-5) [inline .github/workflows/test-linux-cpu.yml:44]:
```suggestion

```

File: .github/workflows/test-linux-gpu.yml [hunk 3]
```diff
@@ -43,7 +43,7 @@ jobs:
         fi
 
         # Create Conda Env
-        conda create -yp ci_env python="${PYTHON_VERSION}" numpy libpng jpeg scipy 'ffmpeg<4.3'
+        conda create -yp ci_env --quiet python="${PYTHON_VERSION}" numpy libpng jpeg scipy  'ffmpeg<4.3'
```
Comment (vfdev-5) [inline .github/workflows/test-linux-gpu.yml:46]:
```suggestion
        conda create -yp ci_env --quiet python="${PYTHON_VERSION}" numpy libpng jpeg scipy 'ffmpeg<4.3'
```

**Comment by NicolasHug** [comment]
Thank you @osalpekar ,

@pmeier just reminded me that he's planning on sharing some of that feedback with the nova team soon. I'll make sure to sync with him so he can report our experience!

**Comment by NicolasHug** [comment]
The 3.10 failure is unrelated. It's a CI/infra issue that needs to be fixed asap - I added it as the first entry in https://github.com/pytorch/vision/issues/7299

**Pull request merged**